Docs: add docs for concatenate transformer

ryantxu opened the pull request:
This adds docs for the concatenate transformer and the new mode for reduce

ryantxu committed:
add docs for concatinate

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -19,26 +19,63 @@ Grafana comes with the following transformations:
-In the **Calculations** field, enter one or more calculation types. Click to see a list of calculation choices. For information about available calculations, refer to the [Calculation list]({{< relref "../calculations-list.md" >}}).
+Consider the input:
+
+Query A:
```
I love examples! Good add.

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -19,26 +19,63 @@ Grafana comes with the following transformations:
+| 2020-07-07 11:34:20 | 6.5     | 15     |
+| 2020-07-07 11:24:20 | 3.2     | 5      |
+
+The reduce transformer has two modes.
```
```suggestion
The reduce transformer has two modes:
- **Series to rows -** Creates a row for each field and a column for each calculation.
- **Reduce fields -** Keeps the existing frame structure, but collapses each field into a single value.
```

oddlittlebird replied:
I suggest separating the definitions from the examples below.

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -19,26 +19,63 @@ Grafana comes with the following transformations:
-Once you select at least one calculation, Grafana reduces the results down to one value using the calculation you select. If you select more than one calculation, then more than one value is displayed.
 
-Here's an example of a table with time series data. Before I apply the transformation, you can see all the data organized by time.
+**Series to rows** will create a row for each field, and a column for each calculation.  For example using the *first* and *last* calculation, 
```
```suggestion
For example, if you used the **First** and **Last** calculation with a **Series to rows** transformation, then 
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -19,26 +19,63 @@ Grafana comes with the following transformations:
+| Errors  | 15      | 5       |
+
+
+**Reduce fields** will keep the existing frame structure, but collapse each field into a single value.  For example with the calculation *last*,
```
```suggestion
The **Reduce fields** with the **Last** calculation,
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -19,26 +19,63 @@ Grafana comes with the following transformations:
+
+
+**Reduce fields** will keep the existing frame structure, but collapse each field into a single value.  For example with the calculation *last*,
+the result would still be two frames, each with one row:
```
```suggestion
results in two frames, each with one row:
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -244,6 +281,34 @@ We would then get :
+
+> **Note:** This transformation is only available in Grafana 7.3+.
+
+This transformation will combine all fields from all frames into one result.  Consider:
```
```suggestion
This transformation combines all fields from all frames into one result.  Consider:
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -244,6 +281,34 @@ We would then get :
+> **Note:** This transformation is only available in Grafana 7.3+.
+
+This transformation will combine all fields from all frames into one result.  Consider:
+
```
```suggestion
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -244,6 +281,34 @@ We would then get :
+| 3.2     | 5      |
+
+
+The result would be:
```
The result of what? In the UI, I see three options to select in the **Name** field. What do each of them do? Which one are you using in this example?

Don't forget to mention the option to add a **Label** when you select the **Add a label with the frame name** option.

oddlittlebird reviewed (changes requested):
Great examples, I have a few suggested edits.

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Merge remote-tracking branch 'origin/master' into concat-docs

ceh committed:
Chore: Require OrgId to be specified in delete playlist command (#29117)

aknuds1 committed:
Drone: Fix Drone config verification for enterprise on Windows (#29118)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ryantxu committed:
DataFrames: add utility function to check if structure has changed (#29006)

* add common flag for knowing if the structure changes

* remove property

* fix test

* fix test

* update comment

* fix jsdoc comments

torkelo committed:
Automation: Adding version bump action

aknuds1 committed:
Chore: Upgrade Go dev tools (#29124)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ceh committed:
Chore: Rewrite preferences test from GoConvey to stdlib and testify (#29129)

torkelo committed:
Automation: Adds new changelog actions (#29142)

* Automation: Adds new changelog actions

* Update

* Add newline

* added link

* Another fix

alexanderzobnin committed:
Provisioning: always pin app to the sidebar when enabled (#29084)

amattheisen committed:
build paths in an os independent way (#29143)

Use filepath instead of path to manipulate filename paths in a way compatible with the target operating system-defined file paths.

domasx2 committed:
respect fronted-logging.enabled flag (#29107)

papagian committed:
Fix authomation text: remove hyphen (#29149)

torkelo committed:
Snapshots: Fixes so that dashboard snapshots show data when using Stat, Gauge, BarGauge or Table panels (#29031)

cinaglia committed:
Fix panic when using complex dynamic URLs in app plugin routes  (#27977)

* remove unused function to interpolate URLs

* share function to add headers between ds/plugin proxies

* stop performing unnecessary plugin setting lookup

* fix bug causing runtime errors when using complex templated URLs

* lower case util functions not used outside of pluginproxy package

* change test URL to a (valid) dummy URL to make intent clearer

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Fix linting issues caught by ruleguard (#28799)

* Chore: Fix linting issues caught by ruleguard

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Improve error check

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
API: Rewrite tests from goconvey (#29091)

* API: Rewrite tests from goconvey

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Remove unused Go code (#28852)

* Chore: Remove more unused Go code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Upgrade Go etc in build images (#29157)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.3.3 (#29159)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.4.0 (#29160)

aknuds1 committed:
Drone: Upgrade build pipeline tool and build image (#29161)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Docs: Fixes what'new menu and creates index page, adds first draft of release notes to docs (#29158)

* Docs: Fixes what'new menu and creates index page, adds first draft of release notes to docs

* Minor tweaks

* Fixed missing letter

* Updates

* Updated

Co-authored-by: Drone <drone@grafana.com>

aknuds1 committed:
Backend: Rename variables for style conformance (#29097)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Commit:
changed link format from MD to HTML (#29163)

azhiltsov committed:
SQL: Define primary key for tables without it (#22255)

The annotation_tag and alert_rule_tag tables did not have
PRIMARY KEY defined what cause problems with migration to
MariaDB with Galera setup with
 innodb_force_primary_key=1
Or MySQL > 8.0.13 with
 sql_require_primary_key=ON

Which can manifest as follows:

MariaDB
Error 1173: This table type requires a primary key

MySQL
ERROR 3750 (HY000): Unable to create or change a table
without a primary key, when the system variable 'sql_require_primary_key' is set.

Extra reading for curious:
https://jfg-mysql.blogspot.com/2017/08/danger-no-pk-with-RBR-and-mariadb-protection.html

Fixes #12971

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

torkelo committed:
Release: Adding release notes for 7.3.3 (#29168)

* Release: Adding release notes for 7.3.3

* Updates

karlatkinson committed:
CloudWatch: added HTTP API Gateway specific metrics and dimensions (#28780)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.3.3 (#29169)

mckn committed:
DashboardLinks: will only refresh dashboard search when changing tags for link. (#29040)

* fixing so we dont run multiple dashboard links searches when changing variables.

* changed so we fetch the list when open the dashboard links dropdown.

* removed verification of unneccesary requests.

aknuds1 committed:
Chore: Upgrade grafana/build-ci-deploy image to latest Go (#29171)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Table: Fix incorrect condtition for rendering table filter (#29165)

aknuds1 committed:
Drone: Fix publish-packages invocation (#29179)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
playlist: Improve test (#29120)

* playlist: Improve test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

bergquist committed:
trace user login and datasource name instead of id (#29183)

id's are not very helpful when debugging a system.

Signed-off-by: bergquist <carl.bergquist@gmail.com>

krvajal committed:
Panels: fix positioning of the header title (#29167)

the menu toggle icon is now absolute positioned and it no longer  affects the centering of the menu title

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.3.3 (#29189)

dprokop committed:
Refactor declarative series configuration to a config builder (#29106)

* Wip: refactor declarative series configuration to a config builder

* Fix plugins initialization

* Config builder reorg and tests

* Typecheck

* Update packages/grafana-ui/src/components/uPlot/context.ts

* Scales config tweak

* Temp disable tests

* Disable some tests temporarily

torkelo committed:
Automation: Adds GitHub release action (#29194)

* Automation: Adds GitHub release action

* Updated wording

* minor update

dprokop committed:
Introduce eslint-plugin-react (#29053)

* Eslint: allign with latest grafana-eslint-config

* fix ts

* Bump @grafana-eslint-congig version

hugohaggmark committed:
Variables: Adds new Api that allows proper QueryEditors for Query variables (#28217)

* Initial

* WIP

* wip

* Refactor: fixing types

* Refactor: Fixed more typings

* Feature: Moves TestData to new API

* Feature: Moves CloudMonitoringDatasource to new API

* Feature: Moves PrometheusDatasource to new Variables API

* Refactor: Clean up comments

* Refactor: changes to QueryEditorProps instead

* Refactor: cleans up testdata, prometheus and cloud monitoring variable support

* Refactor: adds variableQueryRunner

* Refactor: adds props to VariableQueryEditor

* Refactor: reverted Loki editor

* Refactor: refactor queryrunner into smaller pieces

* Refactor: adds upgrade query thunk

* Tests: Updates old tests

* Docs: fixes build errors for exported api

* Tests: adds guard tests

* Tests: adds QueryRunner tests

* Tests: fixes broken tests

* Tests: adds variableQueryObserver tests

* Test: adds tests for operator functions

* Test: adds VariableQueryRunner tests

* Refactor: renames dataSource

* Refactor: adds definition for standard variable support

* Refactor: adds cancellation to OptionPicker

* Refactor: changes according to Dominiks suggestion

* Refactor:tt

* Refactor: adds tests for factories

* Refactor: restructuring a bit

* Refactor: renames variableQueryRunner.ts

* Refactor: adds quick exit when runRequest returns errors

* Refactor: using TextArea from grafana/ui

* Refactor: changed from interfaces to classes instead

* Tests: fixes broken test

* Docs: fixes doc issue count

* Docs: fixes doc issue count

* Refactor: Adds check for self referencing queries

* Tests: fixed unused variable

* Refactor: Changes comments

hugohaggmark committed:
Tests: Adds expects for observables (#28929)

* Tests: Adds expects for observables

* Refactor: renames matcher

* Chore: changed jest version and removed as dev dependency

* Refactor: Changes after PR review

* wip

* Chore: revert

* Chore: Adds jest-matcher-utils

* Chore: fixed merge error

* Tests: Removed inline snapshots as they are env dependent

* Tests: missed a snapshot

* Refactor: Removed type file after PR comments

dprokop committed:
ESlint: React fixes part 1 (#29062)

* Eslint: allign with latest grafana-eslint-config

* fix ts

* Fix react/jsx-key

* Fix react/no-children-prop

* Fix react/jsx-no-target-blank

wbrowne committed:
Permissions: Validate against Team/User permission role update (#29101)

* validate against role field update

* lowercase error string

* make all msgs consistent style

* fix wording

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* sayonara simple json

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Backend: Fix build (#29206)

* Backend: Fix build

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ryantxu committed:
merge master

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -240,6 +279,33 @@ We would then get :
+| 3.2     | 5      |
+
+
+The result would be:
```
```suggestion
After you concatenate the fields, the data frame would be:
```

ryantxu committed:
merge master

ryantxu committed:
Merge remote-tracking branch 'origin/master' into concat-docs

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -16,26 +16,65 @@ Grafana comes with the following transformations:
+The reduce transformer has two modes:
+- **Series to rows -** Creates a row for each field and a column for each calculation.
+- **Reduce fields -** Keeps the existing frame structure, but collapses each field into a single value.
+
```
```suggestion
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -16,26 +16,65 @@ Grafana comes with the following transformations:
+
+
+For example, if you used the **First** and **Last** calculation with a **Series to rows** transformation, then 
+the result would be
```
```suggestion
the result would be:
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -16,26 +16,65 @@ Grafana comes with the following transformations:
+| AQI     | 6.5     | 3.2     |
+| Errors  | 15      | 5       |
+
+
```
```suggestion
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -242,6 +281,33 @@ We would then get :
+| ------- | ------- | ------- | ------ |
+| 15.4    | 1230233 | 3.2     | 5      |
+
+
```
```suggestion
```

oddlittlebird commented on docs/sources/panels/transformations/types-options.md:
```diff
@@ -242,6 +281,33 @@ We would then get :
+| 15.4    | 1230233 | 3.2     | 5      |
+
+
+
```
```suggestion
```

oddlittlebird reviewed (approved):
Looks great! A few minor edits, then good to go.

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu committed:
Update docs/sources/panels/transformations/types-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>
